card-block-mgmt/card-block-mgmt-json-contract.md

```diff
@@ -0,0 +1,167 @@
+  "userNumList": [],
+  "rgSession": 1
+}
+```
+
+### Successful Response (GETPRELOADDATA):
+#### Return list of block override options and current state of each
+
+```json
+{
+	"results": {
+ "maximumDays": 14,
+ "activeCards": [{
+     "name" : "MY WONDERFUL ATM CARD",
+     "last4" : "8537"
+     },
+     {
+     "name" : "MY ONE AND ONLY VISA CARD",
+     "last4" : "2334"
+     }
+     ],
+	"currentStatus": [{
+				"index": "001",
+				"location": "Colorado",
+				"blocked": true,
+				"expiration": null
+			},
+			{
+				"index": "002",
+				"location": "New Mexico",
+				"blocked": true,
+				"expiration": null
+			},
+   {
+				"index": "002",
+				"location": "Arizona",
+				"blocked": false,
+				"expiration": "08/24/20"
+			}
+		],
+		"disclaimerText": ["disclaimer ",
+                     "lines ",
+                     "with ",
+                     "spaces."]
+	}
+}
+```
+
+### Error Responses (GETPRELOADDATA):
+#### Missing or invalid CFG Letterfile:
+
+```json
+{
+  "errorCode": 501,
+  "loggingErrorMessage": "Error processing Letterfile BANNO.CARD.BLOCK.V1.CFG"
+}
+```
+
+#### Ineligible Account based on account or warning type:
+
+```json
+{
+  "errorCode": 502,
+  "loggingErrorMessage": "Ineligible account [warning xx or type xx]"
+}
+```
+
+#### Member has no active, issued Cards:
+
+```json
+{
+  "errorCode": 504,
+  "loggingErrorMessage": "Mbr has no active, issued cards"
+}
+```
+
+## PROCESSDATA STATE
+
+### Client Request  (PROCESSDATA)
+#### The member has made their selections for the PowerOn to process
+
+```json
+
+{
+  "rgState": "PROCESSDATA",
+  "powerOnFileName": "BANNO.CARD.LOC.MGMT.V1.POW",
+  "userChrList":[
+    {"id": 1, "value": "01:082020,02,03:081520"},  /comma delimited list
+    {"id": 2, "value": ""},                        /number with date = overidden
+    {"id": 3, "value": ""},                        /number without date = blocked
+    {"id": 4, "value": ""},                        /all 5 char values can be used
+    {"id": 5, "value": ""}],                       /maximum of 65 elements
+  "userNumList": [],
+  "rgSession": 1
+}
+```
+
+### Successful Response (PROCESSDATA):
+#### Return list or block override options and current state of each
+```json
+{
+	"results": {
+		"success": true,
+	 "updatedStatus": [{
+				"index": "001",
+				"location": "Colorado",
+				"blocked": true,
+				"expiration": null
+			},
+			{
+				"index": "002",
+				"location": "New Mexico",
+				"blocked": true,
+				"expiration": null
+			},
+   {
+				"index": "002",
+				"location": "Arizona",
+				"blocked": false,
+				"expiration": "08/24/20"
+			}
+		]
+	}
+}
+```
+
+### Unsuccessful Response (PROCESSDATA):
+#### Return error code with detail
+```json
+{
+  "errorCode": 503,
+  "loggingErrorMessage": "Error processing request:"+[error]
+}
+```
+
+### Error codes
+| Code   | Description                        |
+|--------|------------------------------------|
+| 500    | System in memo mode                |
```

Comment:
Why not start the error codes from `1`? To avoid confusion with the HTTP status codes.

Reply:
No good reason not to from my perspective. When I came aboard we were using 500's so I just kept going with that. Should we take it one step further and establish a series of error codes which we can consistently move forward with so an error code 1.. or 10.. or 100 for that matter would mean the same thing from one Banno program to the net?

Reply:
Yep I like that.